Build fails when tests are disabled

<!-- If you're not sure whether your problem is a bug, please ask a question at
http://answers.gazebosim.org instead.-->

## Environment
* OS Version: Ubuntu 22.04
* Source or binary build? source, `main` branch

## Description
* Expected behavior:  Build to succeed when building with tests disabled
* Actual behavior: Building with `-DBUILD_TESTING=OFF` fails with 
```
/bin/sh: 1: Python3::Interpreter: not found
make[2]: *** [compiled/CMakeFiles/gz_msgs_msgs.dir/build.make:2752: gz_msgs_gen/gz/msgs/wrench.pb.h] Error 127
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,8 @@ set(GZ_UTILS_VER ${gz-utils2_VERSION_MAJOR})
 gz_find_package(gz-math7 REQUIRED)
 set(GZ_MATH_VER ${gz-math7_VERSION_MAJOR})
 
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
 #--------------------------------------
 # Find if command is available. This is used to enable tests.
 # Note that CLI files are installed regardless of whether the dependency is

